app/data_models/progress_store.py

```diff
@@ -125,7 +135,7 @@ def update_section_status(
 
     def get_section_status(
         self, section_id: str, list_item_id: Optional[str] = None
-    ) -> str:
+    ) -> Union[str, Any]:
```

Comment:
Always a str?

Reply:
`status` in `Progress` class has type `status: Optional[str] = None` so `self._progress[section_key].status` not always returns a string.

Reply:
We should make this `str` and change the `Optional[str]` of `status` from `Progress`, this is because currently we set status to None, then update it in a subsequent call. It would make our lives easier if status is not optional, we can set it to `IN_PROGRESS` when we first create it, then the subsequent call which overwrites can stay as is. So on L162 in progress store we should pass in the status of in_progress when creating it and updating the type hint on `Progress.status` to `str`.

Reply:
So I changed it from `None` to `IN_PROGRESS` when we first create it, it made `status` a `str`. That seems to work well now.